lib/rules/filename-is-valid-accessible-name-matches.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Matches the below:
+ * - `HTMLInputElement` with `type=image`
+ * - Element with `role=img`
```

Comment:
This is confusing as I thought you wanted to match any element with `role=img` and not just make sure the `img` element had the correct role.